### PR TITLE
Migrate from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by disposedtrolley**
This PR migrates the CI tool from Travis to GitHub Actions. We now have a GH Actions Workflow with two jobs:

1. `test-and-lint` which runs `rake` and `rubocop` when PRs are opened or changes are merged into `master`.
2. `spec-live` which runs the authenticated specs if the action is not running in a fork.

Ruby and JRuby versions are handled by [ruby/setup-ruby](https://github.com/ruby/setup-ruby) which includes
support for all minor versions Travis was testing against. I've also added JRuby 9.2 as it's available in this action.

Credentials for the authenticated tests are provided via repository secrets, which are mapped in the workflow to
the same env vars that Travis had previously set.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      - master
+      - spec_live
+
+env:
+  REPO_OWNER: zendesk
+  REPO_NAME: zendesk_api_client_rb
+
+jobs:
+  test-and-lint:
+    runs-on: ubuntu-latest
```
Comment (magnetikonline) [inline .github/workflows/main.yml:19]: Maybe give these a `name:`

```yaml
name: Test and lint
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    branches:
+      - master
+      - spec_live
+
+env:
+  REPO_OWNER: zendesk
+  REPO_NAME: zendesk_api_client_rb
+
+jobs:
+  test-and-lint:
```
Comment (magnetikonline) [inline .github/workflows/main.yml:18]: Maybe give these a `name:`

```yaml
name: Test and lint
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+          ruby-version: ${{ matrix.ruby }}
+      - name: Test and Lint
+        run: |
+          bundle exec rake
+          bundle exec rubocop
+  spec-live:
```
Comment (magnetikonline) [inline .github/workflows/main.yml:41]: Same here:

```yaml
name: Spec live
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        ruby:
+          - 2.3
+          - 2.4
+          - 2.5
+          - 2.6
+          - jruby-9.1
+          - jruby-9.2
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (magnetikonline) [inline .github/workflows/main.yml:31]:
```suggestion
        uses: zendesk/checkout@v2
```

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
```
Comment (magnetikonline) [inline .github/workflows/main.yml:33]:
```suggestion
        uses: zendesk/setup-ruby@v1
```

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  spec-live:
+    if: github.repository == '$REPO_OWNER/$REPO_NAME' && github.repository_owner == $REPO_OWNER
+    env:
+      SPEC_LIVE_USERNAME: ${{ secrets.SPEC_LIVE_USERNAME }}
+      SPEC_LIVE_PASSWORD: ${{ secrets.SPEC_LIVE_PASSWORD }}
+      SPEC_LIVE_ZENDESK_HOST: ${{ secrets.SPEC_LIVE_ZENDESK_HOST }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (magnetikonline) [inline .github/workflows/main.yml:50]:
```suggestion
        uses: zendesk/checkout@v2
```

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
```
Comment (magnetikonline) [inline .github/workflows/main.yml:52]:
```suggestion
        uses: zendesk/setup-ruby@v1
```

**Comment by magnetikonline** [comment]
Looks good @disposedtrolley - just need to ensure actions are under `zendesk/*` based on organisation policy.

File: .github/workflows/main.yml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+on:
+  push:
+    branches:
+      - master
+      - spec_live
+  pull_request:
+    branches:
+      - master
+      - spec_live
```
Comment (nogates) [inline .github/workflows/main.yml:11]: I don't think you need this. We won't be creating pull requests against the `spec_live` branch

File: .github/workflows/main.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
+name: Main
+
+on:
+  push:
+    branches:
+      - master
+      - spec_live
+  pull_request:
+    branches:
+      - master
```
Comment (magnetikonline) [inline .github/workflows/main.yml:10]: I'd say you can simplify this a little - just use `pull_request:` plain.

```suggestion
on:
  pull_request:
  push:
    branches:
      - master
      - spec_live
```

**Comment by magnetikonline** [comment]
@disposedtrolley one minor tweak - otherwise looks solid 👍

**Review by magnetikonline — APPROVED** [review]
:shipit:

**Review by nogates — APPROVED** [review]

**Pull request merged**